README.md

```diff
@@ -29,6 +29,50 @@ For more comprehensive Meson options (`-Doption=value`), including how to compil
 CUTEst can also be installed via the "make" build system based on [ARCHDefs](https://github.com/ralna/ARCHDefs).
 To use this variant, follow the instructions in the CUTEst [wiki](https://github.com/ralna/CUTEst/wiki).
 
+### How to use the static libraries generated by CUTEst
+
+By default, the CUTEst Meson build system compiles static libraries: `libcutest_single.a` and `libcutest_double.a` (as well as `libcutest_quadruple.a` if built with `-Dquadruple=true`).
+These libraries are incomplete by design, they contain undefined symbols corresponding to problem-specific functions that must be provided by the decoded SIF problem.
+
+The package [SIFDecode](https://github.com/ralna/SIFDecode) describes how to decode a SIF problem and generate the corresponding problem library.
+This results in a workflow like the following:
+```shell
+# Linux / FreeBSD
+gfortran -shared -Wl,--whole-archive libcutest_double.a libsif.a -Wl,--no-whole-archive -o libpb.so
+
+# Windows
+gfortran -shared -Wl,--whole-archive libcutest_double.a libsif.a -Wl,--no-whole-archive -o libpb.dll
+
+# macOS
+gfortran -dynamiclib -Wl,-all_load libcutest_double.a libsif.a -Wl,-noall_load -o libpb.dylib
+```
+
+This is the classical approach, where a custom library is built for each decoded SIF problem.
+It has been in use for decades and remains fully supported.
+
+### New approach with shared libraries and trampolines
+
+A modern alternative is now available.
+It is possible to build shared libraries where missing symbols are resolved dynamically using symbol forwarding, via the mechanism in [`cutest_trampoline.f90`](https://github.com/ralna/CUTEst/blob/master/src/tools/cutest_trampoline.f90).
+
+To enable this feature, CUTEst must be built with shared library support:
+```shell
+meson setup builddir -Ddefault_library=shared
+meson compile -C builddir
+meson install -C builddir
+```
+It requires the version `v2.5.0` of CUTEst.
+
+With this approach, there is no need to generate a new combined library for each decoded SIF problem.
+Instead, you dynamically load the decoded problem library at runtime.
+
+The only user-facing addition is the subroutines `load_routines_` and `unload_routines_` load_routines_`,
```

Comment:
The only user-facing addition is the subroutines `load_routines_` and `unload_routines_`,